File: bin/supabase_analyzer.js
// Node.js script za direktan pristup Supabase bazi
const https = require('https');

const SUPABASE_URL = 'https://gjtabtwudbrmfeyjiicu.supabase.co';
const SUPABASE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImdqdGFidHd1ZGJybWZleWppaWN1Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NDc0MzYyOTIsImV4cCI6MjA2MzAxMjI5Mn0.TwAfvlyLIpnVf-WOixvApaQr6NpK9u-VHpRkmbkAKYk';

function makeRequest(path, filters = '') {
  return new Promise((resolve, reject) => {
    const options = {
      hostname: 'gjtabtwudbrmfeyjiicu.supabase.co',
      port: 443,
      path: `/rest/v1/${path}${filters}`,
      method: 'GET',
      headers: {
        'apikey': SUPABASE_KEY,
        'Authorization': `Bearer ${SUPABASE_KEY}`,
        'Content-Type': 'application/json'
      }
    };

    const req = https.request(options, (res) => {
      let data = '';
      res.on('data', (chunk) => {
        data += chunk;
      });
      res.on('end', () => {
        try {
          const parsed = JSON.parse(data);
          console.log(`   🔍 Raw response:`, typeof parsed, Array.isArray(parsed) ? `Array[${parsed.length}]` : 'Object');
          resolve(parsed);
        } catch (e) {
          console.log(`   ❌ Parse error:`, e.message);
          console.log(`   📄 Raw data:`, data.substring(0, 200));
          reject(e);
        }
      });
    });

    req.on('error', (e) => {
      reject(e);
    });

    req.end();
  });
}

async function analyzePayments() {
  console.log('🔍 === ANALIZA NAPLAĆENOG NOVCA U SUPABASE ===');
  console.log('📅 Datum:', new Date().toLocaleDateString('sr-RS'));
  console.log('');

  try {
    const today = new Date();
    const todayStart = new Date(today.getFullYear(), today.getMonth(), today.getDate());
    const todayEnd = new Date(today.getFullYear(), today.getMonth(), today.getDate(), 23, 59, 59);
    
    const startISO = todayStart.toISOString();
    const endISO = todayEnd.toISOString();

    console.log(`🕐 Period: ${todayStart.toLocaleString('sr-RS')} - ${todayEnd.toLocaleString('sr-RS')}`);
    console.log('');

    // 1. DNEVNI PUTNICI
    console.log('📊 1. DNEVNI PUTNICI (tabela: putnici)');
    const dnevniFilters = `?select=naplatioVozac,iznosPlacanja,vremePlacanja,ime&vremePlacanja=gte.${startISO}&vremePlacanja=lte.${endISO}&mesecnaKarta=neq.true&jeOtkazan=neq.true`;
    
    const dnevniPutnici = await makeRequest('putnici', dnevniFilters);
    console.log(`   Broj zapisa: ${dnevniPutnici.length}`);

    let ukupnoDnevni = 0;
    let pazarDnevni = {};
    let placeniDnevni = 0;

    dnevniPutnici.forEach(putnik => {
      const iznos = parseFloat(putnik.iznosPlacanja || 0);
      if (iznos > 0) {
        const vozac = putnik.naplatioVozac || 'Nepoznat';
        ukupnoDnevni += iznos;
        pazarDnevni[vozac] = (pazarDnevni[vozac] || 0) + iznos;
        placeniDnevni++;
        console.log(`     ✅ ${putnik.ime || 'Nepoznato'} -> ${vozac}: ${iznos.toFixed(0)} RSD`);
      }
    });

    console.log(`   📈 Ukupno dnevni: ${ukupnoDnevni.toFixed(0)} RSD (${placeniDnevni} plaćenih)`);
    console.log('');

    // 2. MESEČNI PUTNICI
    console.log('📊 2. MESEČNI PUTNICI (tabela: mesecni_putnici)');
    const mesecniFilters = `?select=vozac,iznosPlacanja,vremePlacanja,putnikIme&aktivan=eq.true&obrisan=eq.false&jePlacen=eq.true&vremePlacanja=gte.${startISO}&vremePlacanja=lte.${endISO}`;
    
    const mesecniPutnici = await makeRequest('mesecni_putnici', mesecniFilters);
    console.log(`   Broj zapisa: ${mesecniPutnici.length}`);

    let ukupnoMesecni = 0;
    let pazarMesecni = {};

    mesecniPutnici.forEach(putnik => {
      const iznos = parseFloat(putnik.iznosPlacanja || 0);
      if (iznos > 0) {
        const vozac = putnik.vozac || 'Nepoznat';
        ukupnoMesecni += iznos;
        pazarMesecni[vozac] = (pazarMesecni[vozac] || 0) + iznos;
        console.log(`     ✅ ${putnik.putnikIme || 'Nepoznato'} -> ${vozac}: ${iznos.toFixed(0)} RSD`);
      }
    });

    console.log(`   📈 Ukupno mesečni: ${ukupnoMesecni.toFixed(0)} RSD`);
    console.log('');

    // 3. UKUPNI REZULTAT
    const ukupnoSvega = ukupnoDnevni + ukupnoMesecni;
    console.log('🏆 === UKUPAN REZULTAT ===');
    console.log(`💰 UKUPNO NAPLAĆENO DANAS: ${ukupnoSvega.toFixed(0)} RSD`);
    console.log(`   - Dnevni putnici: ${ukupnoDnevni.toFixed(0)} RSD`);
    console.log(`   - Mesečni putnici: ${ukupnoMesecni.toFixed(0)} RSD`);
    console.log('');

    // 4. PAZAR PO VOZAČIMA
    console.log('🚗 PAZAR PO VOZAČIMA:');
    const sviVozaci = [...new Set([...Object.keys(pazarDnevni), ...Object.keys(pazarMesecni)])];
    
    sviVozaci.forEach(vozac => {
      const dnevni = pazarDnevni[vozac] || 0;
      const mesecni = pazarMesecni[vozac] || 0;
      const ukupnoVozac = dnevni + mesecni;
      
      if (ukupnoVozac > 0) {
        console.log(`   ${vozac}: ${ukupnoVozac.toFixed(0)} RSD (dnevni: ${dnevni.toFixed(0)}, mesečni: ${mesecni.toFixed(0)})`);
      }
    });

    console.log('');
    console.log('✅ Analiza završena uspešno!');

  } catch (error) {
    console.error('❌ Greška pri analizi:', error.message);
  }
}

analyzePayments();

File: bin/supabase_analyzer.dart
import 'dart:io';
import 'package:flutter/foundation.dart';
import 'package:supabase_flutter/supabase_flutter.dart';
import 'package:intl/intl.dart';

/// 🔍 DIREKTAN QUERY ZA TAČAN IZNOS NAPLAĆENOG NOVCA
/// Jednostavan script koji se direktno povezuje sa Supabase i izvlači podatke
void main() async {
  // Inicijalizuj Supabase
  await Supabase.initialize(
    url: 'https://gjtabtwudbrmfeyjiicu.supabase.co',
    anonKey:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImdqdGFidHd1ZGJybWZleWppaWN1Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NDc0MzYyOTIsImV4cCI6MjA2MzAxMjI5Mn0.TwAfvlyLIpnVf-WOixvApaQr6NpK9u-VHpRkmbkAKYk',
  );

  final client = Supabase.instance.client;

  debugPrint('🔍 === ANALIZA NAPLAĆENOG NOVCA U SUPABASE ===');
  debugPrint('📅 Datum: ${DateFormat('dd.MM.yyyy').format(DateTime.now())}');
  debugPrint('');

  try {
    final now = DateTime.now();
    final danas = DateTime(now.year, now.month, now.day);
    final danasKraj = DateTime(now.year, now.month, now.day, 23, 59, 59);

    debugPrint(
        '🕐 Analiziram period: ${DateFormat('dd.MM.yyyy HH:mm').format(danas)} - ${DateFormat('HH:mm').format(danasKraj)}');
    debugPrint('');

    // 1. DNEVNI PUTNICI - naplaćeni danas
    debugPrint('📊 1. DNEVNI PUTNICI (tabela: putnici)');
    final dnevniQuery = await client
        .from('putnici')
        .select(
            'naplatioVozac, iznosPlacanja, vremePlacanja, mesecnaKarta, jeOtkazan, status, ime')
        .gte('vremePlacanja', danas.toIso8601String())
        .lte('vremePlacanja', danasKraj.toIso8601String())
        .neq('mesecnaKarta', true)
        .neq('jeOtkazan', true);

    debugPrint('   Broj zapisa: ${dnevniQuery.length}');

    double ukupnoDnevni = 0;
    Map<String, double> pazarDnevni = {};
    int placeniDnevni = 0;

    for (final putnik in dnevniQuery) {
      final iznos = (putnik['iznosPlacanja'] ?? 0).toDouble();
      final vozac = putnik['naplatioVozac'] ?? 'Nepoznat';
      final ime = putnik['ime'] ?? 'Nepoznato';

      if (iznos > 0) {
        ukupnoDnevni += iznos;
        pazarDnevni[vozac] = (pazarDnevni[vozac] ?? 0) + iznos;
        placeniDnevni++;
        debugPrint('     ✅ $ime -> $vozac: ${iznos.toStringAsFixed(0)} RSD');
      }
    }

    debugPrint(
        '   📈 Ukupno dnevni: ${ukupnoDnevni.toStringAsFixed(0)} RSD ($placeniDnevni plaćenih)');
    debugPrint('');

    // 2. MESEČNI PUTNICI - naplaćeni danas
    debugPrint('📊 2. MESEČNI PUTNICI (tabela: mesecni_putnici)');
    final mesecniQuery = await client
        .from('mesecni_putnici')
        .select(
            'vozac, iznosPlacanja, vremePlacanja, putnikIme, aktivan, obrisan, jePlacen')
        .eq('aktivan', true)
        .eq('obrisan', false)
        .eq('jePlacen', true)
        .gte('vremePlacanja', danas.toIso8601String())
        .lte('vremePlacanja', danasKraj.toIso8601String());

    debugPrint('   Broj zapisa: ${mesecniQuery.length}');

    double ukupnoMesecni = 0;
    Map<String, double> pazarMesecni = {};

    for (final putnik in mesecniQuery) {
      final iznos = (putnik['iznosPlacanja'] ?? 0).toDouble();
      final vozac = putnik['vozac'] ?? 'Nepoznat';
      final ime = putnik['putnikIme'] ?? 'Nepoznato';

      if (iznos > 0) {
        ukupnoMesecni += iznos;
        pazarMesecni[vozac] = (pazarMesecni[vozac] ?? 0) + iznos;
        debugPrint('     ✅ $ime -> $vozac: ${iznos.toStringAsFixed(0)} RSD');
      }
    }

    debugPrint('   📈 Ukupno mesečni: ${ukupnoMesecni.toStringAsFixed(0)} RSD');
    debugPrint('');

    // 3. UKUPNI REZULTAT
    final ukupnoSvega = ukupnoDnevni + ukupnoMesecni;
    debugPrint('🏆 === UKUPAN REZULTAT ===');
    debugPrint(
        '💰 UKUPNO NAPLAĆENO DANAS: ${ukupnoSvega.toStringAsFixed(0)} RSD');
    debugPrint('   - Dnevni putnici: ${ukupnoDnevni.toStringAsFixed(0)} RSD');
    debugPrint('   - Mesečni putnici: ${ukupnoMesecni.toStringAsFixed(0)} RSD');
    debugPrint('');

    // 4. PAZAR PO VOZAČIMA
    debugPrint('🚗 PAZAR PO VOZAČIMA:');
    final sviVozaci = {...pazarDnevni.keys, ...pazarMesecni.keys}.toSet();

    for (final vozac in sviVozaci) {
      final dnevni = pazarDnevni[vozac] ?? 0;
      final mesecni = pazarMesecni[vozac] ?? 0;
      final ukupnoVozac = dnevni + mesecni;

      if (ukupnoVozac > 0) {
        debugPrint(
            '   $vozac: ${ukupnoVozac.toStringAsFixed(0)} RSD (dnevni: ${dnevni.toStringAsFixed(0)}, mesečni: ${mesecni.toStringAsFixed(0)})');
      }
    }

    debugPrint('');
    debugPrint('✅ Analiza završena uspešno!');
  } catch (e) {
    debugPrint('❌ Greška pri analizi: $e');
    exit(1);
  }

  exit(0);
}
